website/views.py

```diff
@@ -295,20 +295,9 @@ def resolve_guid(guid, suffix=None):
 
     # Retrieve guid data if present, error if missing
     try:
-        base_guid = guid.split('_v')[0]
-        base_guid_obj = Guid.objects.get(_id=base_guid.lower())
-
-        if base_guid_obj.is_versioned:
-            version = guid.split('_v')[1] if len(guid.split('_v')) > 1 else None
-            if version:
-                resource = base_guid_obj.versions.filter(version=version).first().referent
-            else:
-                # redirection to the same route, but with the specified version
-                version = base_guid_obj.versions.order_by('-version').first().version
-                new_guid = f'{base_guid}_v{version}'
-                return redirect(f'/{new_guid}/{suffix}' if suffix else f'/{new_guid}/', code=302)
-        else:
-            resource = Guid.objects.get(_id=base_guid.lower()).referent
+        resource = Guid.load_referent(guid)
```

Comment:
Will it be helpful if we return both resource and a boolean flag `is_versioned`. We can use this extra flag when we need it.

```python
resource, is_versioned = Guid.load_referent(guid)
if is_versioned:
    ...
...
```

```python
resource, _ = Guid.load_referent(guid)
...
```